### PR TITLE
Structure

### PR DESCRIPTION
**Pull request by danialsyed6**
I've made the changes to the structure according to the given format.

### CODE REVIEW

File: src/containers/Dashboard/index.tsx [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import React from 'react';
+
+import { Dashboard as DashboardComponent } from '../../components';
+
+const Dashboard = () => {
+  return <DashboardComponent />;
+};
+
+export default Dashboard;
```
Comment (JavedAhmadDeveloper) [inline src/containers/Dashboard/index.tsx:9]: container function name must be like DashboardContainer

File: src/containers/Dashboard/index.tsx [hunk 2]
```diff
@@ -0,0 +1,9 @@
+import React from 'react';
+
+import { Dashboard as DashboardComponent } from '../../components';
+
```
Comment (JavedAhmadDeveloper) [inline src/containers/Dashboard/index.tsx:4]: folder name must start with small letter like dashboard in your case its Dashboard

File: src/state/index.ts [hunk 3]
```diff
@@ -0,0 +1,16 @@
+import { configureStore } from '@reduxjs/toolkit';
+import createSagaMiddleware from 'redux-saga';
+
+import rootSaga from './ducks';
+import dashboardSlice from './ducks/dashboard/slice';
+
+const saga = createSagaMiddleware();
+
```
Comment (JavedAhmadDeveloper) [inline src/state/index.ts:8]: create a folder inside state for middelware and export saga middelware from there

File: src/state/ducks/student/saga.ts [hunk 4]
```diff
@@ -0,0 +1,16 @@
+import axios from 'axios';
+import { call, put } from 'redux-saga/effects';
+import { getStudentsSuccess } from './slice';
+
+async function getStudents() {
+  const { data } = await axios.get('https://api.thecatapi.com/v1/breeds');
+  data.length = 10;
+  return data;
+}
```
Comment (JavedAhmadDeveloper) [inline src/state/ducks/student/saga.ts:9]: create a folder of utils inside that make fetch api function just like in your provided folder structure and call that function in generator function for api calling

File: src/state/ducks/index.ts [hunk 5]
```diff
@@ -0,0 +1,10 @@
+import { takeLatest } from 'redux-saga/effects';
+import { workGetStudents } from './student/saga';
+
+import { getStudentsRequest } from './student/slice';
+
+function* rootSaga() {
+  yield takeLatest(getStudentsRequest.type, workGetStudents);
+}
+
```
Comment (JavedAhmadDeveloper) [inline src/state/ducks/index.ts:9]: root saga is used to combine all sagas, listen for actions in specific saga..

File: src/state/index.ts [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+const store = configureStore({
+  reducer: { dashboaed: dashboardSlice },
+  middleware: [saga],
+});
+
+saga.run(rootSaga);
+
```
Comment (JavedAhmadDeveloper) [inline src/state/index.ts:15]: create root reducer for combining reducer and dashboard spelling mistake

**Review by IshfaqAhmedAwan — APPROVED** [review]

**Pull request merged**